### ISSUE TITLE
Combining with WiFiManager library doesn't work

### ISSUE THREAD

**Issue by Enterius** [kakopappa/arduino-esp8266-alexa-multiple-wemo-switch#12]
Hi!
I'm trying to combine your precious library with one built in Arduino called WiFiManager. It seems quite easy but whatever I do it doesn't want to work. My serial monitor ends on:
"Begin multicast ..
Udp multicast server started at 239.255.255.250:1900
WebServer started on port: 
90
WebServer started on port: 
91
Adding switches upnp broadcast responder
Adding switch : floor white index : 0
Adding switch : floor color index : 1"

And that's it. It won't respond to Alexa devices discovery. And I have no idea what is wrong, because without WiFiManager it works perfectly. But I need some easier way to configure WiFi parameters than upload a sketch every time. So I would be very grateful if you could spend some time checking what might be wrong with combining those two libraries together. Is it something that I do wrong or is it some problem with libraries incompatibility?

My actual code looks like that:
`// WiFiManager libraries
#include <ESP8266WiFi.h>          //https://github.com/esp8266/Arduino
#include <DNSServer.h>
#include <ESP8266WebServer.h>
#include <WiFiManager.h>          //https://github.com/tzapu/WiFiManager

// Alexa Wemos libraries
// #include <ESP8266WiFi.h>
// #include <ESP8266WebServer.h>
#include <WiFiUdp.h>
#include <functional>
#include "switch.h"
#include "UpnpBroadcastResponder.h"
#include "CallbackFunction.h"

// for manual config start
#define CONFIG_PIN 0

//for LED status
#include <Ticker.h>
Ticker led;

void Blink()
{
  //toggle state
  int state = digitalRead(BUILTIN_LED);  // get the current state of GPIO1 pin
  digitalWrite(BUILTIN_LED, !state);     // set pin to the opposite state
}
// LED status end

// Alexa wemo
// prototypes
boolean connectWifi();

//on/off callbacks
void floorwhiteOn();
void floorwhiteOff();
void floorColorOn();
void floorColorOff();

void floorWhiteOn() {
  digitalWrite(16, LOW);
  Serial.print("Switch 1 turn on ...");
}

void floorWhiteOff() {
  digitalWrite(5, HIGH);
  Serial.print("Switch 1 turn off ...");
}

void floorColorOn() {
  digitalWrite(5, LOW);
  Serial.print("Switch 2 turn on ...");
}

void floorColorOff() {
  digitalWrite(5,HIGH);
  Serial.print("Switch 2 turn off ...");
}

UpnpBroadcastResponder upnpBroadcastResponder;

Switch *floorwhite = NULL;
Switch *floorcolor = NULL;
// Alexa end

//gets called when WiFiManager enters configuration mode
void configModeCallback (WiFiManager *myWiFiManager) {
  Serial.println("Entered config mode");
  Serial.println(WiFi.softAPIP());
  //if you used auto generated SSID, print it
  Serial.println(myWiFiManager->getConfigPortalSSID());
  //entered config mode, make led toggle faster
  led.attach(0.2, Blink);
}

void setup() {
  
  Serial.begin(9600);
  pinMode(BUILTIN_LED, OUTPUT);
  pinMode(16, OUTPUT); // D0
  pinMode(5, OUTPUT); // D1
  pinMode(CONFIG_PIN, INPUT); // set pin 0 (D3) as input for setup mode
  
  // start ticker with 0.5 because we start in AP mode and try to connect
  led.attach(0.6, Blink);

  //WiFiManager
  //Local intialization. Once its business is done, there is no need to keep it around
  WiFiManager wifiManager;
  //reset settings - for testing
  //wifiManager.resetSettings();

  //set callback that gets called when connecting to previous WiFi fails, and enters Access Point mode
  wifiManager.setAPCallback(configModeCallback);

  wifiManager.setConfigPortalTimeout(300); // 5 minutes timeout for setup mode

  //fetches ssid and pass and tries to connect
  //if it does not connect it starts an access point with the specified name
  //and goes into a blocking loop awaiting configuration
  if (!wifiManager.autoConnect("Enterius_Switch")) {
    Serial.println("failed to connect and hit timeout");
    //reset and try again, or maybe put it to deep sleep
    ESP.reset();
    delay(1000);
  }

  //if you get here you have connected to the WiFi
  Serial.println("connected...yeey :)");
  led.detach();
  //keep LED on
  digitalWrite(BUILTIN_LED, LOW);

  // Switches definitions
 
  upnpBroadcastResponder.beginUdpMulticast();

    // Define your switches here. Max 14
    // Format: Alexa invocation name, local port no, on callback, off callback
    floorwhite = new Switch("floor white", 90, floorWhiteOn, floorWhiteOff);
    floorcolor = new Switch("floor color", 91, floorColorOn, floorColorOff);

    Serial.println("Adding switches upnp broadcast responder");
    upnpBroadcastResponder.addDevice(*floorwhite);
    upnpBroadcastResponder.addDevice(*floorcolor);
 
}
  

void loop() {
 
   if ( digitalRead(CONFIG_PIN) == LOW ) {
    led.attach(0.2, Blink);
    //WiFiManager
    //Local intialization. Once its business is done, there is no need to keep it around
    WiFiManager wifiManager;
    wifiManager.startConfigPortal("Enterius_Switch");
    
    
    if (!wifiManager.startConfigPortal("Enterius_Switch")) {
      Serial.println("failed to connect and hit timeout");
      delay(3000);
      //reset and try again, or maybe put it to deep sleep
      ESP.reset();
      delay(5000);
}

}

// Main Alexa Wemos code

upnpBroadcastResponder.serverLoop();
    
    floorcolor->serverLoop();
    floorwhite->serverLoop();

    
  
}
`

I'm just a beginner so maybe I am making some stupid mistake? I spend a few last days on this and it seems that I can't solve this problem by myself. So please, please help.

Thanks and best regards
Chris
